PYXAID2/runMD2.py

```diff
@@ -735,114 +726,114 @@ def runMD(params):
+                    S_soc = ovlp
+
+                    sig1.real().show_matrix("%s/0_sig1_%d_re" % (rd, curr_index) )
+                    sig1.imag().show_matrix("%s/0_sig1_%d_im" % (rd, curr_index) )
+                    sig2.real().show_matrix("%s/0_sig2_%d_re" % (rd, curr_index) )
+                    sig2.imag().show_matrix("%s/0_sig2_%d_im" % (rd, curr_index) )
+                    sig3.real().show_matrix("%s/0_sig3_%d_re" % (rd, curr_index) )
+                    sig3.imag().show_matrix("%s/0_sig3_%d_im" % (rd, curr_index) )
 
                 else:
-
-                    print "multiple k-point for spin-polarized case is not yet implemented"
+                    print "Multiple k-points scheme with SOC is not yet implemented"
                     sys.exit(0)
 
+            
+            # spin-polarized case
+            if  nac_method == 3:
+                # check whether the adiabatic and diabatic basis have the same number of plane waves
+                # the reason why I used the read_qe_wfc_info is because I will need the ngw 
+                # to check the consistency 
+                # But the read_qe_index does not read it, so in order to avoid the changes in the Libra code, 
+                # I use the read_qe_wfc_info.
+                info_wfc0 = QE_methods.read_qe_wfc_info("%s/curr0/x0.export/wfc.1" % wd,0)
+                info_wfc1 = QE_methods.read_qe_wfc_info("%s/curr1/x1.export/wfc.1" % wd,0)
+
+                if info_wfc0["ngw"] != info_wfc1["ngw"]:
+                    print "Error: the number of plane waves between diabatic and adiabatic does not equal"
+                    sys.exit(0)
 
 
+                prms = {"do_orth": 1, "root_directory" : rd, "curr_index" : curr_index, "print_overlaps" : 1, "dt": dt}
+                compute_Hvib(coeff_curr1, coeff_next1, coeff_curr0, coeff_next0, e_curr1, e_next1, e_curr0, e_next0, prms)
 
 
-            H.real().show_matrix("%s/0_Ham_%d_re" % (rd, curr_index) )
-            H.imag().show_matrix("%s/0_Ham_%d_im" % (rd, curr_index) )    
-            S.real().show_matrix("%s/0_S_%d_re" % (rd, curr_index) )
-            S.imag().show_matrix("%s/0_S_%d_im" % (rd, curr_index) )
 
-            if nac_method == 2:
-                H_dia.real().show_matrix("%s/0_Ham_%d_re" % (rd, curr_index) )
-                H_dia.imag().show_matrix("%s/0_Ham_%d_im" % (rd, curr_index) )
+            if nac_method == 0:
+                H.real().show_matrix("%s/0_Ham_%d_re" % (rd, curr_index) )
+                H.imag().show_matrix("%s/0_Ham_%d_im" % (rd, curr_index) )
 
-            if nac_method == 3:
+            if nac_method == 1 or nac_method == 3:
                 H_dia.real().show_matrix("%s/0_H_dia_%d_re" % (rd, curr_index) )
                 H_dia.imag().show_matrix("%s/0_H_dia_%d_im" % (rd, curr_index) )
 
+            if nac_method == 2 or nac_method == 3:
+                H_soc.real().show_matrix("%s/0_H_soc_%d_re" % (rd, curr_index) )
```

Comment:
OK. But there is two kinds of Ham (spin diabatic, spin adiabatic) for nac_method =3. So I think at least H, H_soc needs be introduced.
this leave the user manually define the prefix for vibronic Ham to consider whether the SOC effect is introduced.